src/main/java/com/nonononoki/alovoa/service/ScheduleService.java

```diff
@@ -63,9 +66,23 @@ public void scheduleLong() {
 		if (enableSchedules) {
 			Date date = new Date();
 			cleanContact(date);
+			cleanNonConfirmedUsers(date);
 		}
 	}
 
+	private void cleanNonConfirmedUsers(final Date date) {
+		long ms = date.getTime();
+		ms -= nonConfirmedUsersCleanupDelay;
+		Date d = new Date(ms);
+
+		List<User> users = userRepo.findByConfirmedIsFalse().stream()
```

Comment:
Will add unit-tests once above approach is confirmed.

Reply:
1. Purge, no soft delete
2. No, delete all data
3. I don't think any email notification is needed.

Reply:
@Nonononoki please review once have added unit tests.

Reply:
@lucifercr07  You can use the date filtering directly in the JPA (e.g. `findByConfirmedIsFalseAndDatesCreationDateBefore(Date date))

Reply:
@Nonononoki thanks have addressed the issue.

Reply:
Code is looking good, but why did you need to change the RegisterServiceTest?

Reply:
Wanted to create new mock users as only `testUsers` was getting used across all tests in that file. Also delete users method would throw error since few of the users would be purged and couldn't be found. Please let me know if you feel there's any other effective way to do this would change accordingly.

Reply:
@Nonononoki please let me know if any changes required to be done here.